components/ImageContext.js

```diff
@@ -0,0 +1,37 @@
+import {createContext, useEffect, useState} from "react";
+
+const ImageContext = createContext();
+
+const ImageContextProvider = props => {
+  const [favorites, setFavorites] = useState([]);
+
+  // add function with a proper check if the favorite to be stored is already in your state variable
+  const addFavorite = favorite => {
+    const savedFavorite = favorites.find(fav => fav.id === favorite.id);
+    console.log(savedFavorite);
+    if (savedFavorite) {
```

Comment:
please remove all `console.log()` calls. We don't want that the user can see our debug output.

Reply:
Done.